CLAUDE.md

```diff
@@ -0,0 +1,285 @@
+- `JitsiConnection`, `JitsiConference`, etc. - Main API classes
+
+### TypeScript Migration Guidelines
+- **All new code should be TypeScript** (`.ts` extension)
+- When modifying existing JavaScript files, consider converting to TypeScript
+- Maintain backward compatibility in public API
+- Use strict type checking and avoid `any` type
+- Add proper JSDoc comments.
+
+### Testing Requirements
+- Write tests for new functionality using Jasmine syntax
+- Place test files alongside source with `.spec.js/.spec.ts` extension
+- Test both success and error scenarios
+- Mock external dependencies (XMPP connections, WebRTC APIs)
+- Ensure tests work with both Karma test runner and direct Node.js execution
+
+## Architecture Best Practices
+
+### Event System
+- Use typed events from `/service/` directory
+- Follow EventEmitter pattern consistently
+- Document event parameters and timing in JSDoc comments
+- Emit events for all significant state changes
+
+### Module Design
+- Keep modules self-contained with minimal cross-dependencies
+- Use clear interfaces between XMPP signaling and WebRTC media layers
+- Abstract protocol details behind higher-level APIs
+- Provide consistent error handling and reporting
+
+### WebRTC Integration
+- Use TraceablePeerConnection for enhanced debugging
+- Handle browser compatibility through RTCUtils abstraction
+- Implement proper track lifecycle management
+
+### XMPP Protocol Handling
+- Implement Strophe plugins for protocol extensions
+- Handle connection states and recovery scenarios
+- Support both BOSH and WebSocket transports
+- Maintain protocol compliance with XEP specifications
+
+## Common Development Scenarios
+
+### Adding New WebRTC Features
+1. Implement core functionality in `/modules/RTC/`
+2. Add XMPP signaling support in `/modules/xmpp/`
+3. Expose through main API classes (JitsiConference, JitsiConnection)
+4. Update TypeScript definitions
+5. Add comprehensive tests
+6. Document public API changes
+
+### Protocol Extensions
+1. Create Strophe plugin in `/modules/xmpp/strophe.*.js`
+2. Integrate with existing signaling layer
+3. Add event definitions in `/service/`
+4. Test with both BOSH and WebSocket transports
+5. Ensure backward compatibility
+
+### Quality Control Enhancements
+1. Modify controllers in `/modules/qualitycontrol/`
+2. Update codec preferences and constraints
+3. Test across different network conditions
+4. Verify mobile and desktop compatibility
+5. Monitor performance impact
+
+## Debugging and Troubleshooting
+
+### Common Issues
+- **XMPP connection problems**: Check signaling layer and transport type
+- **WebRTC negotiation failures**: Enable TraceablePeerConnection debugging
+- **E2EE key management**: Verify worker thread communication
+- **Cross-browser compatibility**: Use RTCUtils abstraction layer
+- **Mobile integration**: Ensure proper lifecycle management
+
+### Debug Tools
+- Set `config.debug = true` in jitsi-meet for verbose logging
```

Comment:
Is this even true?

Reply:
good catch, not really, this is related to the tests in jitsi-meet. will remove it.

Reply:
removed the whole section. it didn't feel like something useful to me.